presto-hive/src/main/java/com/facebook/presto/hive/gcs/GcsAccessTokenProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.gcs;
+
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import org.apache.hadoop.conf.Configuration;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class GcsAccessTokenProvider
+        implements AccessTokenProvider
+{
+    public static final String GCS_ACCESS_KEY_CONF = "presto.gcs.oauth-access-token";
+    public static final Long EXPIRATION_TIME_MILLISECONDS = HOURS.toMillis(1);
```

Comment:
let me ask @luohao: do you think we need to make this configurable?

Reply:
Google OAuth token has fixed 1-hour expiration time.

Reply:
2 questions
1. why does the api accept an expiration time? Is that because people use different kinds of tokens.
2. Based on https://github.com/GoogleCloudPlatform/bigdata-interop/blob/61689bdd4e0dcdc3d2063952202f8f0fd7d39622/util-hadoop/src/main/java/com/google/cloud/hadoop/util/AccessTokenProvider.java and https://github.com/GoogleCloudPlatform/bigdata-interop/blob/baf277aa80c983858328729d66b316a8d662e473/util-hadoop/src/main/java/com/google/cloud/hadoop/util/CredentialFromAccessTokenProviderClassFactory.java#L79 it seems like the expiration time is supposed to be a time when the token will expire and not just a duration of how long the token will be active.

Reply:
Hi @rschlussel , @luohao and I discussed about this. In our case, the token might expire after some time, while, we could not support refresh, as the token is passed from user. The expiration time here is not working, it just fulfill the api for access token.
@luohao and I are inclined to not make it configurable, as it does not have effect to control the token's expiration time. @rschlussel what do you think?

Reply:
I'm still trying to understand what the purpose of the time field is in the api and if we're setting it correctly. I think we should make sure to set it correctly for the api (is the time a duration or is it the expiration time?).  If the value we put is actually totally useless, we should set it to some obviously nonsense value (like 0) with a comment that the api field doesn't do anything.

I'm fine not making it configurable if it doesn't make sense to.

Reply:
I think I understand now.  Does the following sound correct: It's used only to refresh the token to say how long it can be refreshed for and we don't support refreshing the token because we don't control it.  In that case, I would set the value to null with a comment.